beacon_node/beacon_chain/src/schema_change/migration_schema_v12.rs

```diff
@@ -0,0 +1,226 @@
+use crate::beacon_chain::{BeaconChainTypes, FORK_CHOICE_DB_KEY, OP_POOL_DB_KEY};
+use crate::persisted_fork_choice::PersistedForkChoiceV11;
+use operation_pool::{PersistedOperationPool, PersistedOperationPoolV12, PersistedOperationPoolV5};
+use slog::{debug, info, Logger};
+use state_processing::{
+    common::get_indexed_attestation, per_block_processing::is_valid_indexed_attestation,
+    VerifyOperation, VerifySignatures,
+};
+use std::sync::Arc;
+use store::{Error, HotColdDB, KeyValueStoreOp, StoreItem};
+
+pub fn upgrade_to_v12<T: BeaconChainTypes>(
+    db: Arc<HotColdDB<T::EthSpec, T::HotStore, T::ColdStore>>,
+    log: Logger,
+) -> Result<Vec<KeyValueStoreOp>, Error> {
+    let spec = db.get_chain_spec();
+
+    // Load a V5 op pool and transform it to V12.
+    let PersistedOperationPoolV5 {
+        attestations_v5,
+        sync_contributions,
+        attester_slashings_v5,
+        proposer_slashings_v5,
+        voluntary_exits_v5,
+    } = if let Some(op_pool) = db.get_item(&OP_POOL_DB_KEY)? {
+        op_pool
+    } else {
+        debug!(log, "Nothing to do, no operation pool stored");
+        return Ok(vec![]);
+    };
+
+    // Load the persisted fork choice so we can grab the state of the justified block and use
+    // it to verify the stored attestations, slashings and exits.
+    let fork_choice = db
+        .get_item::<PersistedForkChoiceV11>(&FORK_CHOICE_DB_KEY)?
+        .ok_or_else(|| Error::SchemaMigrationError("fork choice missing from database".into()))?;
+    let justified_block_root = fork_choice
+        .fork_choice_store
+        .unrealized_justified_checkpoint
+        .root;
+    let justified_block = db
+        .get_blinded_block(&justified_block_root)?
+        .ok_or_else(|| {
+            Error::SchemaMigrationError(format!(
+                "unrealized justified block missing for migration: {justified_block_root:?}",
+            ))
+        })?;
+    let justified_state_root = justified_block.state_root();
+    let mut state = db
+        .get_state(&justified_state_root, Some(justified_block.slot()))?
+        .ok_or_else(|| {
+            Error::SchemaMigrationError(format!(
+                "justified state missing for migraiont: {justified_state_root:?}"
```

Comment:
```suggestion
                "justified state missing for migration: {justified_state_root:?}"
```

Reply:
I'm impressed by how many letters I managed to permute here lol